keycloak/sms-provider/src/main/java/org/sunbird/keycloak/login/phone/AbstractPhoneFormAuthenticator.java

```diff
@@ -1,117 +1,112 @@
-        context.getAuthenticationSession().setAuthNote(AbstractPhoneFormAuthenticator.ATTEMPTED_USERNAME, username);
-
-        UserModel user = null;
-        try {
-            user = getUser(context, username);
-        } catch (ModelDuplicateException mde) {
-            ServicesLogger.LOGGER.modelDuplicateException(mde);
-
-            // Could happen during federation import
-            if (mde.getDuplicateFieldName() != null && mde.getDuplicateFieldName().equals(UserModel.EMAIL)) {
-                setDuplicateUserChallenge(context, Errors.EMAIL_IN_USE, Messages.EMAIL_EXISTS, AuthenticationFlowError.INVALID_USER);
-            } else {
-                setDuplicateUserChallenge(context, Errors.USERNAME_IN_USE, Messages.USERNAME_EXISTS, AuthenticationFlowError.INVALID_USER);
-            }
-
-            return false;
-        }
-
-        if (invalidUser(context, user)) {
-            return false;
-        }
-
-        if (!validatePassword(context, user, inputData)) {
-            return false;
-        }
-
-        if (!enabledUser(context, user)) {
-            return false;
-        }
-
-        String rememberMe = inputData.getFirst("rememberMe");
-        boolean remember = rememberMe != null && rememberMe.equalsIgnoreCase("on");
-        if (remember) {
-            context.getAuthenticationSession().setAuthNote(Details.REMEMBER_ME, "true");
-            context.getEvent().detail(Details.REMEMBER_ME, "true");
-        } else {
-            context.getAuthenticationSession().removeAuthNote(Details.REMEMBER_ME);
-        }
-        context.setUser(user);
-        return true;
+  private static final Logger logger = Logger.getLogger(AbstractPhoneFormAuthenticator.class);
+
+  @Override
+  public boolean validateUserAndPassword(AuthenticationFlowContext context,
+      MultivaluedMap<String, String> inputData) {
+    String username = inputData.getFirst(AuthenticationManager.FORM_USERNAME);
+    logger.debug("AbstractPhoneFormAuthenticator@validateUserAndPassword - Username -" + username);
+
+    if (username == null) {
+      context.getEvent().error(Errors.USER_NOT_FOUND);
+      Response challengeResponse = invalidUser(context);
+      context.failureChallenge(AuthenticationFlowError.INVALID_USER, challengeResponse);
+      return false;
+    }
+
+    // remove leading and trailing whitespace
+    username = username.trim();
+
+    context.getEvent().detail(Details.USERNAME, username);
+    context.getAuthenticationSession()
+        .setAuthNote(AbstractPhoneFormAuthenticator.ATTEMPTED_USERNAME, username);
+
+    UserModel user = null;
+    List<UserModel> userModels = null;
+    try {
+      userModels = SunbirdAuthUtil.getUser(context, username);
+      if (userModels.size() > 1) {
+        context.getEvent().error(Messages.INVALID_USER);
+        Response challenge =
+            context.form().setError("multiple users are associated with this phone.").createLogin();
```

Comment:
Please provide a hint to resolution. The string needs to be extracted so that it can be localised into different languages.

Reply:
It's not definite here that the users were identified by phone. Suggest raising the ModelDuplicateException when multiple users are found with the same phone.

Reply:
Use `setDuplicateUserChallenge` as in lines 78 and 81